### DISCUSSION TITLE
Response - Change Status Code VS Raise raise HTTPException(status_code=202, detail="Still in processing")

### DISCUSSION THREAD

**Discussion by Ben621** [fastapi/fastapi#11002] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, HTTPException

app = FastAPI()

tasks = {"foo": "Listen to the Bar Fighters"}


@app.put("/get-or-create-task/{task_id}", status_code=200)
def get_or_create_task(task_id: str, response: Response):
    if task_id not in tasks:
        response.status_code = status.HTTP_202_ACCEPTED
        return None
    return tasks[task_id]

# OR 

@app.put("/get-or-create-task/{task_id}", status_code=200)
def get_or_create_task(task_id: str):
    if task_id not in tasks:
        raise HTTPException(status_code=202, detail="still in processing")
       
    return tasks[task_id]
```

### Description

Hey,
1. what is the difference between using `response: Response` or  `raise HTTPException(status_code=202, detail="still in processing")` ?
what is the best practices?
2. how to notify the swagger that return 202 when the task is still in processing, and when is finish return the BaseModel?
> note: i'm doing polling model until my  task are finished.

thank you very mcuh

### Operating System

Linux

### FastAPI Version

109

### Pydantic Version

2.5.x

### Python Version

3.11.x

**Comment by n8sty** [comment, marked as answer]
I believe this is what you're looking for: https://fastapi.tiangolo.com/advanced/additional-responses/.

**Reply by Ben621** [reply]
Hey @n8sty 
yes that was i'm looking for about  topic 2.   🆒 
but do you have an idea about topic 1 ?
